src/Symfony/Component/Security/Core/User/OAuth2User.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+/**
+ * UserInterface implementation used by the access-token security workflow with an OIDC server.
+ *
+ * @experimental
+ */
+class OAuth2User implements UserInterface
+{
+    private array $additionalClaims = [];
+
+    public function __construct(
+        private array $roles = ['ROLE_USER'],
+        // Standard Claims (https://datatracker.ietf.org/doc/html/rfc7662#section-2.2)
+        private ?string $scope = null,
+        private ?string $clientId = null,
+        private ?string $username = null,
+        private ?string $tokenType = null,
+        private ?int $exp = null,
+        private ?int $iat = null,
+        private ?int $nbf = null,
+        private ?string $sub = null,
+        private ?string $aud = null,
+        private ?string $iss = null,
+        private ?string $jti = null,
+
+        // Additional Claims ("
+        //    Specific implementations MAY extend this structure with
+        //    their own service-specific response names as top-level members
+        //    of this JSON object.
+        // ")
+        ...$additionalClaims
```

Comment:
could this work as a regular array argument?

Reply:
To be confirmed. As the object is called using the spread operator, this seems to be the safest way.
As it is inspired by the OIDC, the signature of the OIDCUser should also be changed.